script/vendor-mimes

```diff
@@ -25,11 +49,22 @@ data.each do |mime, meta|
   mimes[mime].concat extensions
 end
 
+log_info "Formatting primary hash and writing to file.."
 strlen = mimes.keys.max_by(&:length).length
 output = ""
 output << "# Woah there. Do not edit this file directly.\n"
 output << "# This file is generated automatically by script/vendor-mimes.\n\n"
 mimes = mimes.sort_by { |k,v| k }
 output << mimes.map { |mime,extensions| "#{mime.ljust(strlen)} #{extensions.sort.join(" ")}" }.join("\n")
 
+config = File.expand_path "../lib/jekyll/mime.types", __dir__
 File.write(config, output)
+log_info "Done! See: #{config.inspect.white}"
+
+# --- Generate JSON file from mime-type data ----
+puts
+
+log_info "Dumping sanitized hash to file as JSON.."
+json_file = File.expand_path "../lib/jekyll/commands/serve/mime_types.json", __dir__
+File.write(json_file, JSON.pretty_generate(data) + "\n")
```

Comment:
Saving the whole database means we have superfluous data. That bloats the size of the gem and it increases start-up time. I'd love to see the pre-processing for the file happen here. Filter the DB to a JSON object whose keys are MIME types and the value is the desired charset, like:

```json
{
    "text/plain": "utf-8"
    // ...
}
```

Then in the servlet, you can query the object by the MIME type like you're doing now, but its value is the charset rather than an object potentially containing the charset.

If this doesn't dramatically reduce the size of the json file then we can keep it as-is.

Reply:
I am on board with the idea.
Since the servlet only uses the JSON object to inject the charset, it makes sense to remove the surrounding cruft.